sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -303,18 +317,35 @@ class SQLMetricsSuite extends SharedSparkSession with SQLMetricsTestUtils
     withTempView("testDataForJoin") {
       // Assume the execution plan is
       // ... -> SortMergeJoin(nodeId = 1) -> TungstenProject(nodeId = 0)
-      val leftJoinQuery = "SELECT * FROM testData2 left JOIN testDataForJoin ON " +
+      val query1 = "SELECT * FROM testData2 LEFT JOIN testDataForJoin ON " +
         "testData2.a = testDataForJoin.a"
-      val rightJoinQuery = "SELECT * FROM testDataForJoin right JOIN testData2 ON " +
+      val query2 = "SELECT * FROM testDataForJoin RIGHT JOIN testData2 ON " +
         "testData2.a = testDataForJoin.a"
-
-      Seq((leftJoinQuery, false), (leftJoinQuery, true), (rightJoinQuery, false),
-        (rightJoinQuery, true)).foreach { case (query, enableWholeStage) =>
+      val query3 = "SELECT * FROM testData2 RIGHT JOIN testDataForJoin ON " +
+        "testData2.a = testDataForJoin.a"
+      val query4 = "SELECT * FROM testData2 FULL OUTER JOIN testDataForJoin ON " +
+        "testData2.a = testDataForJoin.a"
+      val boundCondition1 = " AND testData2.b >= testDataForJoin.b"
+      val boundCondition2 = " AND testData2.a >= testDataForJoin.b"
+
+      Seq((query1, 8L, false),
+        (query1 + boundCondition1, 7L, false),
+        (query1 + boundCondition1, 7L, true),
+        (query3 + boundCondition2, 3L, false),
+        (query3 + boundCondition2, 3L, true),
+        (query4, 8L, false),
+        (query4, 8L, true),
+        (query4 + boundCondition1, 7L, false),
+        (query4 + boundCondition1, 7L, true),
+        (query1, 8L, true),
+        (query2, 8L, false),
+        (query2, 8L, true)).foreach { case (query, rows, enableWholeStage) =>
         val df = spark.sql(query)
         testSparkPlanMetrics(df, 1, Map(
           0L -> (("SortMergeJoin", Map(
             // It's 8 because we read 6 rows in the left and 2 row in the right one
```

Comment:
Please update this comments accordingly because this is invalid after this PR.

Reply:
Yes, I removed the comment entirely as there are multiple queries and the comment is no longer valid for all the queries.